main.tf

```diff
@@ -3,7 +3,7 @@ data "aws_vpc" "vpc" {
 }
 
 locals {
-  vpc_name = lookup(data.aws_vpc.vpc.tags, "Name", var.vpc_id)
+  vpc_name = replace(lookup(data.aws_vpc.vpc.tags, "Name", var.vpc_id), ".", "-")
```

Comment:
I would suggest adding a separate local and perform the replace for that local, then use the new local in L46:

```
  name = replace(format("%.255s", lower(replace("tf-redis-${var.name}-${var.env}-${local.vpc_name}-${random_id.salt.hex}", "_", "-"))), "/\\s/", "-")
```

as `vpc_name` is used for other parts and such change would make the descriptions unclear.

And maybe the replace would be honestly better directly in the name, like:

```
name = replace(format("%.255s", lower(replace("tf-redis-${var.name}-${var.env}-replace(${local.vpc_name}, ".", "-")-${random_id.salt.hex}", "_", "-"))), "/\\s/", "-")
```
or some sort of combination, it is worth to play up a bit with it.

Reply:
note: I haven't tested that my proposal is syntactically correct